h5grove/encoders.py

```diff
@@ -104,26 +103,39 @@ def encode(content, encoding: Optional[str] = "json") -> Response:
         - `csv`: nD arrays in downloadable csv files
         - `npy`: nD arrays in downloadable npy files
         - `tiff`: 2D arrays in downloadable TIFF files
+    :param dtype: Data type conversion
+        - `origin` (default): No conversion
+        - `safe`: Convert to a type supported by JS typedarray
+        Only supported for nD array/scalars.
     :returns: A Response object containing content and headers
     :raises ValueError: If encoding is not among the ones above.
     """
+    if dtype in ("origin", None):
+        cast_content = content
+    elif dtype == "safe":
+        cast_content = sanitize_array(content, copy=False)
+    else:
+        raise ValueError(f"Unsupported dtype {dtype}")
```

Comment:
I think this is not correct to call `sanitize_array` here without any checks. `content` can be anything and not only a `ndarray`. 

I think it would be clearer to raise an error if `dtype` was given for a non-array content.

Reply:
Yes I agree, there was no checks whatsoever.